unified-runtime/source/adapters/level_zero/v2/command_buffer.cpp

```diff
@@ -66,12 +66,36 @@ ur_exp_command_buffer_handle_t_::ur_exp_command_buffer_handle_t_(
     ur_context_handle_t context, ur_device_handle_t device,
     v2::raii::command_list_unique_handle &&commandList,
     const ur_exp_command_buffer_desc_t *desc)
-    : commandListManager(
+    : isUpdatable(desc ? desc->isUpdatable : false),
+      isInOrder(desc ? desc->isInOrder : false),
+      commandListManager(
           context, device,
           std::forward<v2::raii::command_list_unique_handle>(commandList),
-          v2::EVENT_FLAGS_COUNTER, nullptr),
-      isUpdatable(desc ? desc->isUpdatable : false), context(context),
-      device(device) {}
+          isInOrder ? v2::EVENT_FLAGS_COUNTER : 0, nullptr, false),
+      context(context), device(device) {}
+
+ur_exp_command_buffer_sync_point_t
+ur_exp_command_buffer_handle_t_::getSyncPoint(ur_event_handle_t event) {
+  syncPoints.push_back(event);
+  return static_cast<ur_exp_command_buffer_sync_point_t>(syncPoints.size() - 1);
```

Comment:
nit: even tough it's unlikely, can we just throw an exception when syncPoints.size() overflows uint32_t?